### PR TITLE
feature: Header is fixed and changes colours

### PR DESCRIPTION
**Pull request by Tulun**
-Uses Intersectional Observer to update the header to light colours once it crosses into a certain section.
-Uses scroll position to make header sticky.

### CODE REVIEW

File: components/pages/Home/Home.tsx [hunk 1]
```diff
@@ -10,19 +11,34 @@ import {
...
+  const eHotItWorks = useIntersectionObserver(howItWorksRef, {
+    threshold: 0.49,
+  });
+  const isIntersectingHowItWorksSection = !!eHotItWorks?.isIntersecting;
   return (
     <Layout>
       <Wrapper>
-        <Header />
+        <Header isIntersecting={isIntersectingLightArea} activeLink={isIntersectingHowItWorksSection ? 0 : -1} />
         <Hero />
```
Comment (Tulun) [inline components/pages/Home/Home.tsx:31]: Question I have here: creating 5 refs to determine active seems a bit hacky. Is there a better way of doing this in a cleaner manner?

File: components/pages/Home/Home.tsx [hunk 2]
```diff
@@ -10,19 +11,34 @@ import {
...
 export function Home() {
+  const headerRef = useRef<HTMLDivElement | null>(null);
+  const howItWorksRef = useRef<HTMLDivElement | null>(null);
+  const eHeader = useIntersectionObserver(headerRef, {
+    threshold: 0.1328,
+  });
+  const isIntersectingLightArea = !!eHeader?.isIntersecting;
+  const eHotItWorks = useIntersectionObserver(howItWorksRef, {
+    threshold: 0.49,
+  });
```
Comment (Tulun) [inline components/pages/Home/Home.tsx:25]: These thresholds seem to give the desired effect but seem very arbitrary, I'd rather it be cleaner.

**Comment by ryanwolhuter** [comment]
Couple notes to help here:

1. I'd recommend using [headroom](https://kyleamathews.github.io/react-headroom/) for showing/hiding the header on scroll — this will take a lot of boilerplate work off you.

2. I don't think you need to use the intersection observer api for the color changing feature. 

For achieving the color changing, all you need to know is the DOM positions of the elements that the header should change color on. The intersection observer api is more intended for situations where _how much_ elements are intersecting also matters.

Here's what I would do.

Create a `HeaderColor` context. Give it state like:

```
lightRegionRefs: {
  someSection: SomeSectionRef
  ...
  }
```

You can populate the context from your layout component. Then you can read those context values in the header. 

Then you can get the y position and height of each of these regions from the ref. Use that to calculate rects in which the color should change.

Finally you can set up a function that checks if the rectangle of the header is within one of those regions, and change the color if so.

One gotcha to keep in mind here for nextjs — the heights and y positions of all the elements will be zero until the page mounts. There's a `useMounted` hook in the helpers that you can use if you need to wait until mount.

File: components/HowItWorks/HowItWorks.tsx [hunk 3]
```diff
@@ -2,17 +2,20 @@ import { useRef, useState } from "react";
...
+  // const refTwo = useRef<HTMLDivElement | null>(null);
   const entryOne = useIntersectionObserver(ref, {});
-  const entryTwo = useIntersectionObserver(ref, {});
+  // const entryTwo = useIntersectionObserver(ref, {});
   const isOneVisible = !!entryOne?.isIntersecting;
 
   const [entryScrollY, setEntryScrollY] = useState(0);
   const [topRedHeight, setRedTopHeight] = useState(0);
-  if (entryOne) console.log(entryOne);
+  // if (entryOne) console.log(entryOne);
```
Comment (daywiss) [inline components/HowItWorks/HowItWorks.tsx:18]: some spare comments

**Review by daywiss — APPROVED** [review]
tested in the app, seems good to me. might want to wait for ryan to review

**Pull request merged**